### ISSUE TITLE
Cannot publish packages that enable features in dependencies (regression)

### ISSUE THREAD

**Issue by hdevalence** [rust-lang/crates.io#1334]
Hi, I'm trying to publish a crate that contains the following in its `Cargo.toml`

```toml
[features]
nightly = ["radix_51", "subtle/nightly", "clear_on_drop/nightly"]
```

to enable `nightly` features in dependencies.  However, it's no longer possible to publish this crate to `crates.io`, since it fails with:
```
error: api errors: invalid upload request: invalid value: string "subtle/nightly", expected a valid feature name at line 1 column 2315
```

This appears to be the error generated by [this code](https://github.com/rust-lang/crates.io/blob/master/src/views/krate_publish.rs#L126), and the same `Cargo.toml` was accepted up until a few days ago, so my guess is that this is a recent regression?

**Comment by CvX** [comment]
This regression has been introduced in PR #1331:
[`Feature` Deserializer](https://github.com/rust-lang/crates.io/blob/4bf9efa/src/views/krate_publish.rs#L121) uses `Crate::valid_feature_name()` which is now `Crate::valid_feature()`.

A quick fix would be to change [src/views/krate_publish.rs:124](https://github.com/rust-lang/crates.io/blob/4bf9efa/src/views/krate_publish.rs#L124), but I'm wondering if the distinction between [Feature](https://github.com/rust-lang/crates.io/blob/4bf9efa/src/views/krate_publish.rs#L53) and [FeatureName](https://github.com/rust-lang/crates.io/blob/4bf9efa/src/views/krate_publish.rs#L55) is necessary? @sgrif?

**Comment by hdevalence** [comment]
Thanks for looking into this! It would be great if it was possible to deploy the fix.

**Comment by fafhrd91** [comment]
when fix should be expected? it is not possible to publish packages with re-export features

**Comment by sgrif** [comment]
:eyes:

**Comment by sgrif** [comment]
Sorry for the delay in looking into this. The fix in #1335 is correct. I've added a regression test to #1336, and will get it deployed once it's merged.

> I'm wondering if the distinction between Feature and FeatureName is necessary?

It is. A feature is what can appear in `foo = { version = "*", features = [...] }`. It is between 1 and 2 feature names separated by a slash (technically a crate name followed by a feature name, but that's the same in this case). A feature name is what can appear in `[features]` as a key. It cannot contain a slash.

> It would be great if it was possible to deploy the fix.

I've made sure everyone who can deploy was pinged.
